src/main/java/seedu/address/model/delivery/DeliveryDatePredicate.java

```diff
@@ -0,0 +1,39 @@
+package seedu.address.model.delivery;
+
+import java.util.function.Predicate;
+
+/**
+ * Tests if a {@code Delivery}'s date matches the given date.
+ */
+
+public class DeliveryDatePredicate implements Predicate<Delivery> {
+    private final DateTime targetDate;
+
+    /**
+     * Creates a DeliveryDatePredicate instance based on the given targetDateTime
+     * @param targetDateTime DateTime as specified by user used to filter deliveries
+     */
+    public DeliveryDatePredicate(DateTime targetDateTime) {
+        this.targetDate = targetDateTime;
+    }
+
+    @Override
+    public boolean test(Delivery delivery) {
+        DateTime deliveryDate = delivery.getDeliveryDate();
```

Comment:
Should this be abstracted into a method inside the delivery class to check for equal dates instead in consideration of the Law of Demeter

Reply:
Yes thank you for pointing this out, I have made the changes.